### DISCUSSION TITLE
realtime not working locally or in prod

### DISCUSSION THREAD

**Discussion by amilkov3** [supabase/supabase#13953] [Questions]
so i have 2 issues running realtime
1) at supabase cli version 1.50.11, supabase realtime image is started as part of supabase start but then no docker port is exposed (i believe it should just be 4000)...is this intentional. i get a bunch of failures when attempting to --exclude realtime and then cloning the realtime repo and manually running the docker compose but id rather not do all that since its more of a nuisance to develop locally. is there some simple way to expose 4000:4000 with args to supabase start or something?
2) supabase sockets in prod arent doing anything. i double checked my server side code is executing:
```
const realtime = new RealtimeClient(
        process.env.NEXT_PUBLIC_SUPABASE_URL!.includes('localhost')
            ? 'ws://localhost:4000/socket'
            : `wss://${process.env.NEXT_PUBLIC_SUPABASE_PROJ_REF}.supabase.co/realtime/v1`,
        {
            params: { apikey: process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY },
        },
    );
realtime.connect()
const x = realtime.channel('onchain').subscribe();
await x.send({
type: 'broadcast',
event: 'onchain',
payload: { agreementId: agree.id },
});
```
but i see nothing in the realtime logs in the web studio and the client definitely isnt receiving any events

**Comment by teorema-community** [comment]
i have the same problem
